app/jobs/create_derivatives_job.rb

```diff
@@ -22,4 +23,14 @@ def parent_needs_reindex?(file_set)
     return false unless file_set.parent
     file_set.parent.thumbnail_id == file_set.id
   end
+
+  def reload(af_object)
+    return af_object unless af_object.persisted?
+
+    adapter = Hyrax.config.valkyrie_metadata_adapter
```

Comment:
If this is the same code as in FileSetActor, I would recommend pulling it out of both into a service (e.g. `lib/wings/services/af_object_reloader_service.rb`).